### PR TITLE
Allow wizard to scan for a Device ID to use

### PR DESCRIPTION
**Pull request by uzlonewolf**
Screenshot from 2023-04-16 12-55-17

### CODE REVIEW

**Comment by uzlonewolf** [comment]
With the "associated-users" API endpoint getting blocked I can see an argument where an empty DeviceID should also kick off a scan, but I do hope we can eventually find another way to get the device list without needing a device ID.

**Comment by jasonacox** [comment]
I wonder why they would restrict this? My only thought is that there could be an attack vector they are trying to harden a bit (e.g. bad actor trying to fetch all of your devices after somehow gaining access somehow to your IoT API credentials).

File: tinytuya/wizard.py [hunk 1]
```diff
@@ -138,6 +138,18 @@ def wizard(color=True, retries=None, forcescan=False, nocloud=False, quicklist=F
         with open(DEVICEFILE, "r") as infile:
             tuyadevices = json.load( infile )
     else:
+        if 'apiDeviceID' in config and config['apiDeviceID'] == 'scan':
```
Comment (jasonacox) [inline tinytuya/wizard.py:141]: Should we make that `config['apiDeviceID'].lower() == 'scan':` to ignore user entered caps?

Comment (uzlonewolf) [reply tinytuya/wizard.py]: Done!

**Comment by uzlonewolf** [comment]
I have no idea.

I was looking through the "official" Python API implementation at https://github.com/tuya/tuya-iot-python-sdk/blob/main/example/api.py and they get the device list by calling `/v1.0/iot-03/users/assets` followed by `/v1.0/iot-02/assets/{}/devices`, but of course they return "permission deny" as well.  Looking at the "Quick Start Guide" at https://developer.tuya.com/en/docs/iot/quick-start1?id=K95ztz9u9t89n I wonder if there's an API Service we need to add authorization for to get access to these.

**Comment by uzlonewolf** [comment]
Going a bit off topic here, but while looking through that repository I noticed `PROTOCOL_DEVICE_REPORT = 4` and thought "hmmm, 'protocol: 4' looks familiar."  A bit more digging reveals while 4 and 20 are the only protocols defined there, https://github.com/tuya/tuya-connect-kit-for-mqtt-embedded-c/blob/without-submodules/include/mqtt_service.h lists a bunch more.  So, it turns out `protocol: 4` means `dev -> cloud push dp data` and `protocol: 5` is `cloud -> dev send dp data`.  Mystery from #308 solved!

**Comment by jasonacox** [comment]
Wow! Amazing find!!  I wonder how we could capture this info gold?  Maybe add it to https://github.com/jasonacox/tinytuya/discussions/260 ?

PR merged. Love the scan function!  I'll push this as v1.12.4 as a minor patch.

**Pull request merged**